update: rewrite the outdated section on avoiding duplicate publications (#3947)
JetBrains/kotlin-web-site: docs/topics/multiplatform/multiplatform-publish-lib.md

Specify the group, version, and the [repositories](https://docs.gradle.org/current/userguide/publishing_maven.html#publishing_maven:repositories) 
where the library should be published. The plugin creates publications automatically.

> > Here and further down the page all examples show the `shared/build.gradle.kts` file, not the root project configuration.

Reviewer: I think it would be better to repeat this in every section with a code sample  (note as a note, but as an action), for example:
> In the `shared/build.gradle.kts` file, specify the group, version..

> This means you don't have to customize your build by adding an empty artifact to the root module of your library to meet the repository's requirements.
>
{type="note"}

The `kotlinMultiplatform` publication may also need the sources and documentation artifacts if that is required by the repository. In that case, 
add those artifacts by using [`artifact(...)`](https://docs.gradle.org/current/javadoc/org/gradle/api/publish/maven/MavenPublication.html#artifact-java.lang.Object-) 
in the publication's scope.

## Host requirements

Except for Apple platform targets, Kotlin/Native supports cross-compilation, allowing any host to produce needed artifacts.

> So, to avoid any issues:

Reviewer: ```suggestion
To avoid any issues during publication:
```

* Publish only from an Apple host when your project targets Apple operating systems.

> * Never publish from multiple hosts to avoid duplicating publications at the repo.

Reviewer: ```suggestion
* Publish all artifacts from one host only to avoid duplicating publications in the repository.
```

  
  Maven Central, for example, explicitly forbids duplicate publications and fails the process. <!-- TBD: add the actual error -->
  

> ### Problems solved since Kotlin 1.9.20 {initial-collapse-state="collapsed"}

Reviewer: I think this whole section could be rephrased to be more actionable, rather than a historic reference, like this:

### If you use Kotlin 1.9.0 and earlier

Before Kotlin 1.9.20, the Kotlin/Native compiler didn't support all the cross-compilation options. If you use these earlier versions, modules in multiplatform projects can be published more than once, resulting in duplicate publications in the repository.

...

To avoid that, assign a main host for each target and check for it like this:

...

We recommend upgrading to the latest version of Kotlin and using the current approach:..

---
Also, the indentation is off in the code block

Did you intentionally removed the Groovy option?

Author: Actionable is good indeed. Rewrote stuff. Also, "1.9.20" was fine to signify something _from before_; to rewrite as you suggested, I asked for a more precise version number.

I did remove the Groovy option because it seemed unnecessary as an example of what not to do, but in this revision, it's back again.
